container/openejb-core/src/main/java/org/apache/openejb/resource/AutoConnectionTracker.java

```diff
@@ -58,16 +60,18 @@
     private final TransactionManager txMgr;
     private final Logger logger = Logger.getInstance(LogCategory.OPENEJB_CONNECTOR, "org.apache.openejb.resource");
     private final ConcurrentMap<ManagedConnectionInfo, ProxyPhantomReference> references = new ConcurrentHashMap<>();
-    private final ReferenceQueue referenceQueue = new ReferenceQueue();
+    private final ReferenceQueue<ConnectionInvocationHandler> referenceQueue = new ReferenceQueue<>();
     private final ConcurrentMap<Class<?>, Class<?>> proxies = new ConcurrentHashMap<>();
     private final ConcurrentMap<Class<?>, Class<?>[]> interfaces = new ConcurrentHashMap<>();
 
     private final boolean cleanupLeakedConnections;
+    private final ExecutorService executorService;
```

Comment:
When using a service, the service should be shut down. I'm not sure right now where the best place for that is, but I'd still like to see the call somewhere.

Reply:
There is new commit that handles this.